src/app.py

```diff
@@ -697,6 +720,127 @@ def render_form(**kwargs):
+        return render_form()
+
+    # If the user has their Github verified
+    if request.method == "POST" and "github_login" in session and "auto_submit" in request.form:
+
+        try:
+            # Gather necessary data
+            create_pr_data = {
+                "file_path": f"data/{request.values['project_name']}.yaml",
+                "file_content": form.get_yaml(),
+                "branch": f"add-project-{request.values['project_name']}",
+                "message": f"Add Project {request.values['project_name']}",
+                "committer": GithubUser.from_token(session["github_login"]['access_token']),
+                "fork_repo": GitHubRepoAPI(GitHubAuth("osg-doc-bot", app.config["OSG_BOT_GITHUB_TOKEN"]), 'osg-doc-bot', 'topology'),
+                "root_repo": GitHubRepoAPI(GitHubAuth("osg-doc-bot", app.config["OSG_BOT_GITHUB_TOKEN"]), 'opensciencegrid', 'topology'),
+            }
+
+            create_pr_response = create_file_pr(**create_pr_data)
+
+            form.clear()
+
+            return render_form(pr_url=create_pr_response['html_url'])
+
+        except GithubReferenceExistsException as error:
+            return render_form(error="A Pull-Request for this project already exists, change Project Name and try again.")
+
+        except GithubRequestException as error:
+            return render_form(error="Unexpected Error: Please submit manually.")
+
+    # Start Github Oauth Flow
+    if request.method == "POST" and "auto_submit" in request.form:
+
+        session['form_data'] = form.as_dict()
+        return redirect(f"/github/login?scope=user:email read:user")
+
+    # Generate the yaml for manual addition
+    if request.method == "POST" and "manual_submit" in request.form:
+
+        form.yaml_output.data = form.get_yaml()
+        return render_form(form_complete=True)
+
+    return render_form()
+
+
+@app.route("/github/login", methods=["GET"])
+def github_login():
+    """Used to login to Github and return to previous location"""
+
+    if "code" in request.args:
+        if request.args['state'] != session.pop("state"):
+            return Response("Github returned incorrect state. Try again or email support@osg-htc.org for support.", 401)
+
+        data = {
+            "code": request.args['code'],
+            "client_id": "2bf9c248a619961eb14a",
+            "client_secret": app.config["GITHUB_OAUTH_CLIENT_SECRET"],
+            "redirect_uri": request.args['redirect_uri']
+        }
+
+        redirect_data = {}
+        try:
+            response = requests.post("https://github.com/login/oauth/access_token", data=data, headers={"Accept": "application/json"})
+            session['github_login'] = response.json()
+
+        except Exception as error:
+            redirect_data['error'] = f"Could not complete exchange with Github for Token: {error}"
+
+        return redirect(
+            f"{urllib.parse.urlparse(request.args['redirect_uri']).geturl()}?{urllib.parse.urlencode(redirect_data)}"
```

Comment:
I know the cool kids are all about the f-strings these days, but when you have big expressions like these embedded in the interpolation parts, it's really difficult to parse visually to see what is string vs code.

Compare:

```suggestion
            "{}?{}".format(
                urllib.parse.urlparse(request.args['redirect_uri']).geturl(),
                urllib.parse.urlencode(redirect_data)
            )
```

Or even `"%s?%s" % (..., ...)`, if you are down with the primordial syntax.

Anyway food for thought for you, here and other such places.

f-strings are fine when you want to do `f"foo={foo} bar={bar}"` - this is supposed to enhance readability. But when f-strings get to be mostly code it loses that.